### ISSUE TITLE
Question on Training Phase

### ISSUE THREAD

**Issue by LeeHoang2710** [jaeseokbyun/RTD#2]
Hi Authors,

Thank you for releasing this great work and the accompanying codebase. I am currently reproducing the results and ran into an issue during the training phase that I hope you can clarify.

According to the paper and repo, the encoder is trained for 2000 epochs. To monitor training quality, I validated the newly trained encoder on the *dress* sub-folder of the FashionIQ dataset during training. However, the recall scores do not seem to improve over time. Instead, they fluctuate within a narrow range and remain much lower than the results reported in your paper.

Below are the validation logs I collected (Recall@10 and Recall@50):

<img width="1138" height="427" alt="Image" src="https://github.com/user-attachments/assets/ee540d87-5aff-4d9c-9b65-04a2a96c041e" />

The performance appears to stagnate and does not show the expected upward trend.

My questions are:

1. **During your training, did you observe clear improvements on CIRR validation dataset over time?**  
2. **Are there additional training details or hyperparameters (e.g., learning rate schedule, temperature scaling, sampling strategy) that are critical for the improvements shown in your paper?**  
3. **Any insight on why validation might plateau like this would be greatly appreciated.**

Thank you very much for your time and for providing this great resource.

**Comment by jaeseokbyun** [comment]
Thanks for your questions.

First, as we clarified in Section 4.1, we select the final model **based on the CIRR validation score**. Could you share which backbone you used? I think you should compare the results with the table in the appendix. Note that the numbers in the main table are averages across all subsets in FashionIQ, whereas the appendix provides per-subset results.


Answer to Question 1:
As mentioned in Remark 4.2, our text encoder is already pre-trained, and the primary objective during training is to learn the relationships between texts. Because of this, we often observe substantial improvements even within the early iterations. This does not hold for every subset, but in many cases, the best-performing model appears relatively early in training. In our experiments, CIRR validation typically improves noticeably during the early stages, while additional iterations do not always lead to better performance (strong improvement in early stage and then show plateau)


Answer to Question 2:
You may try using a smaller learning rate (e.g., 1e-6) or increasing the frequency of validation to better track early improvements.


Answer to Question 3:
This is likely related to the explanation provided in Answer 1.

**Comment by LeeHoang2710** [comment]
Thank you very much for your detailed response!

I am currently using ViT-L/14 LinCIR with noise scale = 0.1, learning rate = 1e-5 and batch size = 64. With this configuration, my reproduced results are very close to the validation numbers in your appendix.

The reason I opened the issue is that during training, I noticed that the model already reaches its best performance around 400–500 epochs, and extending to 2000 epochs does not seem to provide further improvement. Since your explanation clarified that the model improves strongly in the early stage and then plateaus, this behavior now makes sense.

I would like to ask one more question:
Is there a specific reason you chose 2000 epochs for training?
Was this based on experiments, or was it more of a safe upper bound rather than a necessary number of iterations?

Lastly, I’m very interested in this research direction—especially zero-shot composed image retrieval. If you are still actively exploring this topic, I would be very happy to discuss or exchange ideas with you.

Thank you again for sharing your insights and for maintaining this great project!

**Comment by jaeseokbyun** [comment]
I'm glad to hear that the replies are helpful and that you were able to reproduce the results successfully!

Regarding your additional question, it's closer to the latter case. Since our method is a post-hoc approach applicable to diverse ZS-CIR methods across various backbones (as demonstrated in our paper), we set a large maximum training iteration count. Given that training remains efficient even at 2,000 iterations, we chose this value for convenience and to ensure performance across different settings.

Lastly, I'm happy to chat about the research direction! Even though I'm not currently actively exploring this specific topic, I'm always open to chatting and exchanging ideas, feel free to reach out anytime.

Thanks again for your interest in our work!
